### PR TITLE
PLF-8055: Allow enable Login History by configuration

### PR DESCRIPTION
**Pull request by wkhessairi**
This fix adds a system property `exo.loginHistory.enabled` which allows to enable/disable login history data storage

### CODE REVIEW

File: exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryListener.java [hunk 1]
```diff
@@ -45,15 +45,18 @@ public LoginHistoryListener(LoginHistoryService loginHistoryService) throws Exce
    */
   @Override
   public void onEvent(Event<ConversationRegistry, ConversationState> event) throws Exception {
-    String userId = event.getData().getIdentity().getUserId();
-    try {
-      long now = System.currentTimeMillis();
-      if (now - loginHistoryService.getLastLogin(userId) > 180000) {
-        loginHistoryService.addLoginHistoryEntry(userId, now);
-        LOG.info("User " + userId + " logged in.");
+    String enableLoginHistory = System.getProperty("exo.loginHistory.enabled", "true");
```
Comment (thomasdelhomenie) [inline exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryListener.java]: It should be injected in init params in the service LoginHistoryService IMO and exposed with the getter (isEnabled()). It makes it easier to unit test.
And, speaking about unit test, can you add one please ? ;)

Comment (thomasdelhomenie) [inline exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryListener.java]: The property name should be in a constant IMO. And I would rename it to something like "exo.audit.login.enabled" (I expect we will have more auditing capabilities in the future ;) ). WDYT ?

Comment (wkhessairi) [reply exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryListener.java]: yes, ok

Comment (wkhessairi) [reply exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryListener.java]: Why not ! anyway you have a closer view of the future than me ;)

File: exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryServiceImpl.java [hunk 2]
```diff
@@ -27,10 +28,16 @@
  */
 
 public class LoginHistoryServiceImpl implements LoginHistoryService {
+  private boolean enableLoginHistory;
```
Comment (thomasdelhomenie) [inline exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryServiceImpl.java]: We should set the boolean to true to make sure it is enabled by default even if the init param is missing.

Comment (wkhessairi) [reply exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryServiceImpl.java]: yes. I guessed this :+1:

**Comment by thomasdelhomenie** [comment]
@wkhessairi you improved the test coverage, you can raise the ratio ! ;)

**Comment by wkhessairi** [comment]
> @wkhessairi you improved the test coverage, you can raise the ratio ! ;)

done. However I thought about users who used the old implementation of LoginHistoryService. Shouldn't we keep the default constructor of [LoginHistoryService](https://github.com/exodev/platform/blob/develop/exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryServiceImpl.java#L32), and deprecate it perhaps.

**Comment by thomasdelhomenie** [comment]
Can't it cause issue ? Which one will be used by the container ? I think that's fine to keep the PR as is.

**Comment by wkhessairi** [comment]
Okay

**Pull request merged**